src/neo/Ledger/Blockchain.cs

```diff
@@ -517,6 +520,19 @@ private void SendRelayResult(IInventory inventory, VerifyResult result)
         private void UpdateCurrentSnapshot()
         {
             Interlocked.Exchange(ref currentSnapshot, GetSnapshot())?.Dispose();
+            HashSet<UInt160> whiteList = new HashSet<UInt160>
+            {
+                NativeContract.NEO.GetCommitteeAddress(currentSnapshot),
+                GetConsensusAddress(NativeContract.NEO.GetNextBlockValidators(currentSnapshot)),
+                GetConsensusAddress(NativeContract.RoleManagement.GetDesignatedByRole(currentSnapshot, Role.Oracle, currentSnapshot.Height)),
+                GetConsensusAddress(NativeContract.RoleManagement.GetDesignatedByRole(currentSnapshot, Role.StateValidator, currentSnapshot.Height))
+            };
+            Interlocked.Exchange(ref addressWhiteList, whiteList)?.Clear();
```

Comment:
move this code to `RoleManagement` contract?

Reply:
I don't think so.